### PR TITLE
feat(project-selector): Update non member projects UI

### PR DESCRIPTION
**Pull request by lynnagara**
Groups non member projects in the project selector dropdown. Currently
only affects superusers.

### CODE REVIEW

**Comment by lynnagara** [comment]
Doesn't look great, having a hard time making the group label look good with the fixed virtualized height (ideally it would be smaller)

Screen Shot 2019-04-17 at 3 42 04 pm

**Comment by lynnagara** [comment]
Updated design

Screen Shot 2019-04-19 at 2 05 34 pm

**Comment by lynnagara** [comment]
Updated design

Screen Shot 2019-04-19 at 2 06 46 pm

File: src/sentry/static/sentry/app/components/projectSelector.jsx [hunk 1]
```diff
@@ -182,14 +194,47 @@ class ProjectSelector extends React.Component {
...
-    const projectList = sortArray(projects, project => {
-      return [!project.isBookmarked, project.name];
-    });
+    const [projects, nonMemberProjects] = this.getProjects();
 
-    const hasProjects = projectList && !!projectList.length;
+    const hasProjects = projects && !!projects.length;
     const hasProjectWrite = access.has('project:write');
 
+    const getProjectItem = project => ({
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/projectSelector.jsx]: Should this be a method?

**Review by billyvg — COMMENTED** [review]
What happens with search when you only have a member project (and vice versa)?

**Comment by lynnagara** [comment]
Ok, fixed up a bug in non-member selection. Search hasn't changed in behaviour at all (i.e. still the following - if you have "All projects" selected  it always defaults to searching projects with memberships only, if want to search a project you're not a member of you need to explicitly select it on the dropdown

File: src/sentry/static/sentry/app/components/projectSelector.jsx [hunk 2]
```diff
@@ -34,6 +35,9 @@ class ProjectSelector extends React.Component {
     multiProjects: PropTypes.arrayOf(
       PropTypes.oneOfType([PropTypes.string, SentryTypes.Project])
     ),
+    nonMemberProjects: PropTypes.arrayOf(
+      PropTypes.oneOfType([PropTypes.string, SentryTypes.Project])
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/projectSelector.jsx]: When would this prop be a string?

Comment (lynnagara) [reply src/sentry/static/sentry/app/components/projectSelector.jsx]: Ah, it won't be. Updated to only accept Project

**Review by markstory — APPROVED** [review]
Looks good to me.

**Pull request merged**